Sources/PrimerSDK/Classes/Core/PrimerHeadlessUniversalCheckout/Managers/VaultManager.swift

```diff
@@ -0,0 +1,826 @@
+        
+        // MARK: Public functions
+        public override init() {
+            PrimerInternal.shared.sdkIntegrationType = .headless
+            PrimerInternal.shared.intent = .checkout
+            
+            let sdkEvent = Analytics.Event(
+                eventType: .sdkEvent,
+                properties: SDKEventProperties(
+                    name: "\(Self.self).\(#function)",
+                    params: nil))
+            Analytics.Service.record(events: [sdkEvent])
+            
+            super.init()
+        }
+        
+        public func configure() throws {
+            try self.validate()
+        }
+        
+        private func validate() throws {
+            guard PrimerAPIConfigurationModule.decodedJWTToken != nil,
+                  PrimerAPIConfigurationModule.apiConfiguration != nil
+            else {
+                let err = PrimerError.uninitializedSDKSession(userInfo: nil, diagnosticsId: UUID().uuidString)
+                ErrorHandler.handle(error: err)
+                throw err
+            }
+            
+            guard let customerId = PrimerAPIConfigurationModule.apiConfiguration?.clientSession?.customer?.id else {
+                let err = PrimerError.invalidClientSessionValue(
+                    name: "customer.id",
+                    value: nil,
+                    allowedValue: "string",
+                    userInfo: nil,
+                    diagnosticsId: UUID().uuidString)
+                ErrorHandler.handle(error: err)
+                throw err
+            }
+        }
+        
+        public func fetchVaultedPaymentMethods(completion: @escaping (_ vaultedPaymentMethods: [PrimerHeadlessUniversalCheckout.VaultedPaymentMethod]?, _ error: Error?) -> Void) {
+            let vaultService = VaultService()
+            
+            firstly {
+                vaultService.fetchVaultedPaymentMethods()
+            }
+            .done {
+                self.vaultedPaymentMethods = AppState.current.paymentMethods.compactMap({ $0.vaultedPaymentMethod })
+                DispatchQueue.main.async {
+                    completion(self.vaultedPaymentMethods, nil)
+                }
+            }
+            .catch { err in
+                DispatchQueue.main.async {
+                    completion(nil, err)
+                }
+            }
+        }
+        
+        public func deleteVaultedPaymentMethod(id: String, completion: @escaping (_ error: Error?) -> Void) {
+            guard let vaultedPaymentMethod = self.vaultedPaymentMethods?.first(where: { $0.id == id }) else {
+                let err = PrimerError.invalidVaultedPaymentMethodId(
+                    vaultedPaymentMethodId: id,
+                    userInfo: ["file": #file, "class": "\(Self.self)", "function": #function, "line": "\(#line)"],
+                    diagnosticsId: UUID().uuidString)
+                ErrorHandler.handle(error: err)
+                
+                DispatchQueue.main.async {
```

Comment:
Would it make sense to return `error` as `completion(error)` instead of passing it to `primerDidFailWithError`?